### DISCUSSION TITLE
Detecting the scroll bar visibility in a container?

### DISCUSSION THREAD

**Discussion by MuongKimhong** [Textualize/textual#3949] [Q&A] [answered]
By default, container doesn't show the scrollbar unless the contents inside container are longer or bigger than container. Is there anyway to detect that once scroll bar appears on screen?

Currently i'm doing this way
```python
def move_down():
    if mycontainer.vertical_scrollbar.styles.visibility == "visible":
        # scroll bar has appeared, do something 
```

but this one it always give me "visible" even no scroll bar appeared on screen?.

**Comment by TomJGooding** [comment]
The `visibility` style determines whether a widget is visible or not, nothing to do with scrollbars. You probably need the 
[`show_horizontal_scrollbar`](https://textual.textualize.io/api/widget/#textual.widget.Widget.show_horizontal_scrollbar) and/or [`show_vertical_scrollbar`](https://textual.textualize.io/api/widget/#textual.widget.Widget.show_vertical_scrollbar) attributes,

**Reply by MuongKimhong** [reply]
My bad! I meant 
mycontainer.vertical_scrollbar.styles.visibility

Anyway, I tried show_vertical_scrollbar but it always return False no matter scrollbar appears on screen or not.

**Reply by TomJGooding** [reply, marked as answer]
Are you sure that the scrollbar is appearing on your container and not another parent container or the screen?

For example, this seems to work after a quick test:

```python
from textual.app import App, ComposeResult
from textual.containers import VerticalScroll
from textual.widgets import Static


class ExampleApp(App):
    CSS = """
    Static {
        border: tall cornflowerblue;
    }
    """

    def compose(self) -> ComposeResult:
        yield VerticalScroll()

    def on_mount(self) -> None:
        self.set_interval(0.5, self.add_widget)

    def add_widget(self) -> None:
        container = self.query_one(VerticalScroll)
        container.mount(Static("Does the container have a scrollbar yet?"))

        if container.show_vertical_scrollbar is True:
            self.notify("Yes!", timeout=0.5)
        else:
            self.notify("Not yet.", timeout=0.5)


if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```

**Comment by davep** [comment]
Bit late to this party, but if I'm reading the question correctly, it's how to detect when one or both of the scrollbars appear? If so keep in mind that `show_*_scrollbar` are reactives, and so can be watched. For example, here's an example I wrote for someone who asked how they could make it so a widget could only receive focus *if* scrolling was happening:

```python
from textual import on
from textual.app import App, ComposeResult
from textual.containers import VerticalScroll
from textual.widgets import Button, Label

class MaybeFocus(VerticalScroll, can_focus=False):

    def watch_show_vertical_scrollbar(self) -> None:
        self.can_focus = self.show_horizontal_scrollbar or self.show_vertical_scrollbar

    def watch_show_horizontal_scrollbar(self) -> None:
        self.can_focus = self.show_horizontal_scrollbar or self.show_vertical_scrollbar

class FocusWhenScrollApp(App[None]):

    CSS = """
    Label {
        height: 5;
        background: red;
        width: 1fr;
    }
    """

    def compose(self) -> ComposeResult:
        yield Button("MORE!")
        yield MaybeFocus()

    @on(Button.Pressed)
    def more(self) -> None:
        self.query_one(MaybeFocus).mount(Label("Stuff"))

if __name__ == "__main__":
    FocusWhenScrollApp().run()
```
